datalad/distribution/install.py

```diff
@@ -310,90 +162,144 @@ def __call__(
                                 git_clone_opts=git_clone_opts,
-                                annex_opts=annex_opts,
-                                annex_init_opts=annex_init_opts
+                                annex_init_opts=annex_init_opts,
+                                **common_kwargs
                             )
-
-                            installed_items += assure_list(result)
-                        except Exception:
-                            # Note: We don't exactly know what was skipped but
-                            # the `path` requested to be installed, since it will be
-                            # resolved only within the recursive call of install.
-                            lgr.info("Installation of {0} skipped.".format(p))
-
-                    if len(installed_items) == 1:
-                        return installed_items[0]
+                installed_items += assure_list(result)
+
+            if to_get:
+                lgr.debug("Install passes into get %d items", len(to_get))
+                # all commented out hint on inability to pass those options
+                # into underlying install-related calls.
+                # Also need to pass from get:
+                #  annex_get_opts
+                content_by_ds = Get.__call__(
+                    to_get,
+                    # description=description,
+                    # if_dirty=if_dirty,
+                    # save=save,
+                    # git_clone_opts=git_clone_opts,
+                    # annex_init_opts=annex_init_opts
+                    _return_datasets=True,
+                    **common_kwargs
+                )
+
+                # compose content_by_ds into result
+                for dspath in content_by_ds:
+                    ds_ = Dataset(dspath)
+                    if ds_.is_installed():
+                        installed_items.append(ds_)
                     else:
-                        return installed_items
-            else:
-                path = path[0]
+                        lgr.warning("%s was not installed", ds_)
+
+            return installed_items[0] \
+                if len(installed_items) == 1 else installed_items
+
+        if source and path and len(path) > 1:
+            raise ValueError(
+                "install needs a single PATH when source is provided.  "
+                "Was given mutliple PATHs: %s" % str(path))
+
+        # parameter constraints:
+        if not source:
+            raise InsufficientArgumentsError(
+                "a `source` is required for installation")
+
+        # code below deals with a single path only
+        path = path[0] if path else None
+
+        if source == path:
+            # even if they turn out to be identical after resolving symlinks
+            # and more sophisticated witchcraft, it would still happily say
+            # "it appears to be already installed", so we just catch an
+            # obviously pointless input combination
```

Comment:
FTR: code from this point and below is primarily from you and @mih, so I might not be the best person to answer, but let me try.

This is a `UC6_3` in our spreadsheet. ATM it is `save -d. existingds` (verified -- works).  I would prefer us supporting `install -d. existingds` to register a dataset within its super and that is how it supposedly is in master (tried -- it does it but with a side-effect of also saving another modified dataset) because then it would be inline with e.g. how "install -d. -s URL PATH" works -- it does add it to the dataset BUT again with a side-effect of saving unrelated changes! (that is a bug, need to address now)